### ISSUE TITLE
Need Way to "Sanitize" Version

### ISSUE THREAD

**Issue by EarthCitizen** [nebula-plugins/nebula-release-plugin#151]
When the version contains `+`, this causes various problems downstream. See #143 for examples of problems caused. There is a need to be able to remove/replace special characters in some circumstances, like when the version is used in a JAR name.

**Comment by rpalcolea** [comment]
Hi, 

Looking at #143 

This issue is related to the fact that docker doesn't support `+` in tags for example.

```
 docker tag my-sample-image 1.0.0-snapshot.11+eb31c52
Error parsing reference: "1.0.0-snapshot.11+eb31c52" is not a valid repository/tag: invalid reference format
``` 

While we could add a feature to sanitize the version, this would become something like: `1.0.0-snapshot.11.eb31c52` instead but it will be part of version sorting and potentially bring bad results.

What are your thoughts on a feature flag to disable the build metadata instead?

**Comment by EarthCitizen** [comment]
@rpalcolea We use `devSnapshot` and rely on the branch name (or feature name in our case) to be in the version to identify artifacts. I am not sure what the majority of people need, but in our specific case, removing the metadata completely would make it difficult for us to identify artifacts. I recognize, though, that this may not be the use case of most users.

**Comment by rpalcolea** [comment]
Hi! sorry for the delay, we've been busy on this side.

We released `11.0.0` with a flag to sanitize the version. Please refer to https://github.com/nebula-plugins/nebula-release-plugin#sanitize-versions for its usage.

**Comment by EarthCitizen** [comment]
@rpalcolea Thank you!
